perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -783,11 +785,20 @@ public UserExtSource getUserExtSourceFromMultipleIdentifiers(PerunSession sess,
 	public User getUserByExtSourceInformation(PerunSession sess, PerunPrincipal principal) throws UserExtSourceNotExistsException, UserNotExistsException, ExtSourceNotExistsException {
 		String shibIdentityProvider = principal.getAdditionalInformations().get(ORIGIN_IDENTITY_PROVIDER_KEY);
 		if(shibIdentityProvider != null && extSourcesWithMultipleIdentifiers.contains(shibIdentityProvider)) {
-			UserExtSource ues = getUserExtSourceFromMultipleIdentifiers(sess, principal);
-			return getUserByUserExtSource(sess, ues);
-		} else {
-			return getUserByExtSourceNameAndExtLogin(sess, principal.getExtSourceName(), principal.getActor());
+			try {
+				UserExtSource ues = getUserExtSourceFromMultipleIdentifiers(sess, principal);
+				return getUserByUserExtSource(sess, ues);
+			} catch (UserExtSourceNotExistsException ex) {
+				if (!lookupUserByIdentifiersAndExtSourceLogin) {
+					throw new UserExtSourceNotExistsException(ex);
```

Comment:
Nit: we can simply rethrow the existing exception instead of a new one. Creating a new exception can be an expensive operation (if happening too often) but I don't think it would make a difference in this case.
```suggestion
					throw ex;
```